github-process-readme.md

```diff
@@ -0,0 +1,132 @@
+# FAQ – GITHUB Process 
+
+# 1. Configure ssh keys 
+
+A) Generate new ssh keys 
+```sh
+$ ssh-keygen -o -t rsa -b 4096 -C "<seagate-email-address>"  
+```
+B) Update github settings with this public ssh key
+  - Get id_rsa.pub key (default location: /root/.ssh/id_rsa.pub) 
+
+  - Go to github ssh key setting: https://github.com/settings/keys 
+
+  - Add new ssh key and then select Enable SSO option and Authorize for this key
+
+# 2. Git configuration process 
+
+A) Configure git 
+```sh
+$ git config --global user.name ‘Your Name’ 
+
+$ git config --global user.email ‘Your.Name@seagate.com’ 
+
+$ git config --global color.ui auto 
+```
+
+B) Clone s3server  
+
+a) using ssh: `$ git clone git@github.com:Seagate/cortx-s3server.git` 
+
+C) Update the source code:  `$ git submodule update --init  --recursive` 
+
+D) Work on feature branch  
+
+1)  create new branch: `$ git checkout –b dev/SS/test1` 
+
+2)  update your code change and add it to git:  
+                           
+             $ vim README.md 
+
+             $ git diff README.md 
+
+             $ git add README.md 
+
+3)  commit your code change: `$ git commit –m “<JIRA ID>:S3:<info about change>” `
+
+4)  check git log: `$ git log –2 (to see last 2 commits)` 
+
+5)  push your code change:  `$ git push origin dev/SS/test1` (output shows pull request url)
+
+E)  Create pull request for feature branch 
+
+A) Use pull URL showed in prev push command and raise pull requests. 
+
+<img src="images/image1.PNG">
+
+Or  
+
+B) Use GitHub console,  
+
+On GitHub, navigate to the main page of the repository.   
+In the "Branch" menu, choose the branch that contains your commits. 
+
+<img src="images/image2.PNG">
+
+C) Add Reviewers, comments for your pull requests 
+
+<img src="images/image3.PNG">
+
+D) Trigger pre-merge jenkins job using commit id of your change 
+
+1) get commit id of your change 
+
+<img src="images/image4.PNG">
+
+2) start pre-merge jobs using commit id: 
+http://eos-jenkins.mero.colo.seagate.com/job/S3server/job/s3-github-test/ 
```

Comment:
This will not be applicable for cortx-test

Reply:
@seagate-sarang-sawant will do this in the next set of updates. Thanks!